src/app/components/team/Languages/AddLanguageAction.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { graphql, commitMutation } from 'react-relay/compat';
+import { Store } from 'react-relay/classic';
+import Button from '@material-ui/core/Button';
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import ConfirmProceedDialog from '../../layout/ConfirmProceedDialog';
+import languagesList from '../../../languagesList';
+
+function submitAddLanguage({ input, onCompleted, onError }) {
+  commitMutation(Store, {
+    mutation: graphql`
+      mutation AddLanguageActionUpdateTeamMutation($input: UpdateTeamInput!) {
+        updateTeam(input: $input) {
+          team {
+            id
+            get_languages
+          }
+        }
+      }
+    `,
+    variables: {
+      input,
+    },
+    onCompleted,
+    onError,
+  });
+}
+
+const useStyles = makeStyles(theme => ({
+  autocompleteWrapper: {
+    width: theme.spacing(40),
+    height: theme.spacing(12),
+  },
+}));
+
+const AddLanguageAction = ({
+  languages,
+  team,
+}) => {
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [selectedLanguage, setSelectedLanguage] = React.useState('');
+  const classes = useStyles();
+
+  const options = Object.keys(languagesList).filter(code => !languages.includes(code));
+  const getOptionLabel = code => `${languagesList[code].nativeName} (${code})`;
+
+  const handleSelectLanguage = (e, value) => {
+    setSelectedLanguage(value);
+  };
+
+  const handleSubmit = () => {
+    const onCompleted = () => {
+      setSelectedLanguage('');
+      setDialogOpen(false);
+    };
+    const onError = () => {};
+
+    submitAddLanguage({
```

Comment:
I'm using the `isSaving`/`CircularProgress` pattern here. As with the statuses, we don't have individual permissions for languages so it could be a bit tricky to block menu actions on optimistic data.